app/bundles/EmailBundle/Tests/Controller/EmailMapStatsControllerTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\EmailBundle\Tests\Controller;
+
+use Doctrine\DBAL\Exception;
+use Doctrine\ORM\Exception\ORMException;
+use Doctrine\ORM\OptimisticLockException;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\EmailBundle\Controller\EmailMapStatsController;
+use Mautic\EmailBundle\Entity\Email;
+use Mautic\EmailBundle\Model\EmailModel;
+use Mautic\UserBundle\Entity\Role;
+use Mautic\UserBundle\Entity\User;
+use PHPUnit\Framework\MockObject\MockObject;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+class EmailMapStatsControllerTest extends MauticMysqlTestCase
+{
+    private MockObject $emailModelMock;
+
+    private EmailMapStatsController $mapController;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->emailModelMock = $this->createMock(EmailModel::class);
+        $this->mapController  = new EmailMapStatsController($this->emailModelMock);
```

Comment:
Could you please test with actual HTTP routes instead of creating the controller service? Such test confirms that it will work for users. This is testing just a part of the code.

Reply:
This wasn't resolved.

Reply:
Sorry, I missed this comment, but the HTTP request test was added https://github.com/mautic/mautic/pull/12570/files#diff-bce4d2fe9fecbcc0637f5507fc9c522f7124827d238290e460a334316194e345R81